README.md

```diff
@@ -197,7 +197,7 @@ list.tail                    # => Hamster.list(2, 3)
 To add to a list, you use `List#cons`:
 
 ``` ruby
-original = Hamster.list(1, 2, 3)
+original = Hamster::List(1, 2, 3)
```

Comment:
This should be brackets: `Hamster::List[1, 2, 3]`

This shows the awkward situation in which `Vector` `Hash` `Set` are classes (and therefore instances are constructed with `#new`), but `List` is a module... Maybe topic for another discussion.